base/gmp.jl

```diff
@@ -37,27 +37,33 @@ else
 end
 
 
+immutable _withsize end
+
 type BigInt <: Integer
     alloc::Cint
     size::Cint
     d::Ptr{Limb}
-    function BigInt()
-        b = new(zero(Cint), zero(Cint), C_NULL)
-        ccall((:__gmpz_init,:libgmp), Void, (Ptr{BigInt},), &b)
-        finalizer(b, _gmp_clear_func)
-        return b
+    function BigInt(::_withsize, nbits)
+        z = new(Cint(0), Cint(0), C_NULL)
+        # WARNING: nbits == 0 is unsafe, some mpz_* functions expect an abs size >= 1
+        nbits > 0 && ccall((:__gmpz_init2,:libgmp), Void, (Ptr{BigInt}, Culong), &z, nbits)
```

Comment:
I left this as an option for the programmer to manually initialize the `BigIng` object at a later stage. Note that as long as the object (equal to `big(0)`) is not used in a mutable GMP function, this should not be unsafe.